src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaFactory.java

```diff
@@ -54,12 +32,13 @@ private Properties getProps() {
     }
 
     public Consumer<String, String> getConsumer() {
-        return new KafkaConsumer<>(getProps());
+        return new KafkaConsumer<>(kafkaLocationManager.getKafkaProperties());
     }
 
     public SimpleConsumer getSimpleConsumer() {
-        final String[] split = kafkaAddress.split(":");
-        return new SimpleConsumer(split[0], Integer.parseInt(split[1]), 1000, 64 * 1024, "leaderlookup");
+        for (Broker kafkaBroker : kafkaLocationManager.getKafkaBrokers()) {
+            return new SimpleConsumer(kafkaBroker.host, kafkaBroker.port, 1000, 64 * 1024, "leaderlookup");
```

Comment:
I believe in java it is considered a bad practice to directly access class fields outside of the class.